enclave_manager/avalon_enclave_manager/avalon_enclave_bridge.py

```diff
@@ -178,32 +179,49 @@ def get_enclave_basename():
 
 
 # -----------------------------------------------------------------
-def verify_enclave_info(enclave_info, mr_enclave, originator_public_key_hash):
+def verify_enclave_info(worker_mode, enclave_info, mr_enclave, extended_data):
     """
     Verifies enclave signup info
+    - worker_mode - Worker mode. Supported modes are Singleton Enclave,
+                    KME Enclave, WPE Enclave
     - enclave_info is a JSON serialised enclave signup info
       along with IAS attestation report
-    - mr_enclave is enclave measurement value
+    - mr_enclave - enclave measurement value
+    - extended_data - MRENCLAVE value of associated WPE
+      in case of KME Enclave and Verification key of associated KME
+      incase of WPE Enclave
     """
     return enclave.VerifyEnclaveInfo(
-        enclave_info, mr_enclave, originator_public_key_hash)
+        worker_mode, enclave_info, mr_enclave, extended_data)
 
 
 # -----------------------------------------------------------------
-def create_signup_info(originator_public_key_hash, nonce):
+def create_signup_info(nonce, worker_mode=WorkerMode.SINGLETON_ENCLAVE.value,
+                       extended_data="", extended_data_signature="",
+                       kme_attestation=""):
```

Comment:
what is nonce used for ?

Reply:
Nonce is added as part of the IAS request to verify attestation as distinguishing factor.

Reply:
Your answer about the nonce parameter should be added as a @param in the docstring.

Reply:
Sure, will add.